src/features/code_actions.zig

```diff
@@ -366,6 +367,231 @@ fn handleVariableNeverMutated(builder: *Builder, actions: *std.ArrayListUnmanage
     });
 }
 
+fn handleUnorganizedImport(builder: *Builder, actions: *std.ArrayListUnmanaged(types.CodeAction)) !void {
+    const tree = builder.handle.tree;
+    if (tree.errors.len != 0) return;
+
+    const imports = try getImportsDecls(tree, builder.arena);
+
+    // if (std.sort.isSorted(ImportDecl, imports.items, tree, ImportDecl.lessThan)) return;
+
+    const sorted_imports = try builder.arena.dupe(ImportDecl, imports.items);
+    std.sort.heap(ImportDecl, sorted_imports, tree, ImportDecl.lessThan);
+
+    var edits = std.ArrayListUnmanaged(types.TextEdit){};
+
+    // add sorted imports
+    {
+        var new_text = std.ArrayListUnmanaged(u8){};
+        var writer = new_text.writer(builder.arena);
+
+        for (sorted_imports, 0..) |import_decl, i| {
+            if (i != 0 and ImportDecl.addSeperator(sorted_imports[i - 1], import_decl)) {
+                try new_text.append(builder.arena, '\n');
+            }
+
+            try writer.print("{s}\n", .{offsets.locToSlice(tree.source, import_decl.getLoc(tree, false))});
+        }
+        try writer.writeByte('\n');
+
+        const insert_pos: types.Position = pos: {
+            const tokens = tree.tokens.items(.tag);
+            if (tokens.len < 2 or tokens[0] != .container_doc_comment) {
+                break :pos .{ .line = 0, .character = 0 };
+            }
+            // Insert after doc comment
+            break :pos offsets.tokenToPosition(tree, 1, builder.offset_encoding);
```

Comment:
In my limited testing I did on this PR, I have encountered an issue with files that had top-level doc-comments.
Doc comments are emitted per-line. If a top-level doc-comments spans across multiple lines then `getSourceStartIndex` will return the position after the first line.

This should resolved the issue:
```suggestion

        const tokens = tree.tokens.items(.tag);
        const first_token = std.mem.indexOfNone(std.zig.Token.Tag, tokens, &.{.container_doc_comment}) orelse tokens.len;
        const insert_pos = offsets.tokenToPosition(tree, @intCast(first_token), builder.offset_encoding);
```

This should also have test coverage.